### PR TITLE
[4.0] Workflow new article stage category change

### PR DESCRIPTION
**Pull request by bembelimen**
### Summary of Changes
This PR solves a lot of feature requests for the workflow:
- it resets the default workflow stage to "unpublished", so when copying an article (batch, multilingual) the article is unpublished as default.
- When creating a new article you'll see now the inital stage in the dropdown, thanks to @dgrammatiko and this PR #23414 when changing the category, the stage field updates its options.
- Additional the first level of transitions are available, so when creating a new article one can choose to execute the first transition and publish etc. the article without the need of double saving. This simulates the J! 3.x behavior.

### Testing Instructions
Create a new article and change the category, test with multiple workflows and transitions.

@wilsonge

### CODE REVIEW

File: administrator/components/com_categories/Field/CategoryeditField.php [hunk 1]
```diff
@@ -351,9 +351,9 @@ protected function getInput()
 
 		$data['options']             = $this->getOptions();
 		$data['allowCustom']         = $this->allowAdd;
-		$data['customFields']        = (boolean) $this->element['custom-fields-enabled'];
-		$data['customFieldsCatId']   = (string) $this->element['custom-fields-cat-id'];
-		$data['customFieldsSection'] = (string) $this->element['custom-fields-section'];
+		$data['refreshPage']         = (boolean) $this->element['refresh-enabled'];
+		$data['refreshCatId']        = (string) $this->element['refresh-cat-id'];
+		$data['refreshSection']      = (string) $this->element['refresh-section'];
```
Comment (Quy) [inline administrator/components/com_categories/Field/CategoryeditField.php]: Remove extra spaces and align `=`.

**Comment by wilsonge** [comment]
Much better!

File: administrator/components/com_categories/Field/CategoryeditField.php (hunk 1 above)
Comment (bembelimen) [reply administrator/components/com_categories/Field/CategoryeditField.php]: Thx

**Comment by HLeithner** [comment]
With this PR we change the default behavior from published to default unpublished.

Is this really intended? I think every person using Joomla know per default a new article is published.
In my opinion if you copy an article it shouldn't touch the state, especially if I set it by me self before I save2copy.

**Comment by micker** [comment]
no its more mogicial to keep it unpublished like module duplicate and more ...
when you copy item its for modification, if state is published everybody can see a bad article

**Comment by bembelimen** [comment]
If you want article published by default, you can just change the default stage with one click.

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on ba9e30e1e9c66365e6a7d5666db1d42de2acd567<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23458">issues.joomla.org/tracker/joomla-cms/23458</a>.</sub>

**Comment by Hackwar** [comment]
I have tested this item :white_check_mark: successfully on ba9e30e1e9c66365e6a7d5666db1d42de2acd567<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23458">issues.joomla.org/tracker/joomla-cms/23458</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23458">issues.joomla.org/tracker/joomla-cms/23458</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**